### PR TITLE
short circuiting the code and avoid log noise

### PR DESCRIPTION
**Pull request by dloucasfx**
address https://github.com/signalfx/signalfx-agent/issues/1625

Signed-off-by: Dani Louca <dlouca@splunk.com>

### CODE REVIEW

**Review by keitwb — APPROVED** [review]

**Review by keitwb — CHANGES_REQUESTED** [review]
Make the log message at debug level.

File: pkg/core/hostid/aws.go [hunk 1]
```diff
@@ -25,6 +25,10 @@ func AWSUniqueID(cloudMetadataTimeout timeutil.Duration) string {
 		return ""
 	}
 	defer resp.Body.Close()
+	if resp.StatusCode == http.StatusNotFound {
+		log.Info("HTTP status 404, assuming not on EC2")
```
Comment (keitwb) [inline pkg/core/hostid/aws.go:29]: Debug level. This will cause confusion at info level.

Comment (dloucasfx) [reply pkg/core/hostid/aws.go:29]: I don't see it much different from the previous message.
If we move to debug then we potentially undid the previous fix of exposing errors without debug

Comment (keitwb) [reply pkg/core/hostid/aws.go:29]: I'm not really concerned with a real AWS metadata service returning a 404, that would be a very strange scenario.

Comment (dloucasfx) [reply pkg/core/hostid/aws.go:29]: Fair enough! it's just those corner cases that always come back to bite us.
I will push the change

**Review by keitwb — APPROVED** [review]

**Comment by keitwb** [comment]
This is superseded by #1633

**Pull request closed without merging**